ProcessOut/Classes/ProcessOut.swift

```diff
@@ -365,14 +382,96 @@ public class ProcessOut {
+                        // Cancel the timeout as we catched the redirect
+                        timeOutHandler.cancel()
+                        if error != nil {
+                            handler.onError(error: error!)
+                        } else {
+                            // Finggerprint token successfuly received, we continue the authorization flow
+                            makeCardPayment(invoiceId: invoiceId!, token: token!, handler: handler, with: with)
+                        }
+                    }), forURLScheme: appURLScheme)
+                    // Add the webview to the app view
+                    webView = WKWebView(frame: with.view.frame, configuration: configuration)
+                    webView.load(URLRequest(url: url))
+                    webView.isHidden = true
+                    with.view.addSubview(webView)
+                    
+                    // Start the timeout handler with a 10s timeout
+                    DispatchQueue.main.asyncAfter(deadline: .now() + 10.0, execute: timeOutHandler)
+                } else {
+                    // Fallback on earlier versions
+                    fallbackFingerprint(invoiceId: invoiceId, URL: customerAction.value,handler: handler, with: with)
+                }
+                
+                break
             }
         } else {
             handler.onSuccess(invoiceId: invoiceId)
         }
     }
+
+    public static func continueThreeDSVerification(invoiceId: String, token: String, completion: @escaping (String?, ProcessOutException?) -> Void) {
+        let authRequest = AuthorizationRequest(source: token)
+        if let body = try? JSONEncoder().encode(authRequest) {
+            do {
+                let json = try JSONSerialization.jsonObject(with: body, options: []) as! [String : Any]
+                HttpRequest(route: "/invoices/" + invoiceId + "/authorize", method: .post, parameters: json, completion: {(data, error) -> Void in
+                    if data != nil {
+                        completion(invoiceId, nil)
+                    } else {
+                        completion(nil, error!)
+                    }
+                })
+            } catch {
+                completion(nil, ProcessOutException.GenericError(error: error))
+            }
+        } else {
+            completion(nil, ProcessOutException.InternalError)
+        }
+    }
+    
+    private static func fallbackFingerprint(invoiceId: String, URL: String, handler: ThreeDSHandler, with: UIViewController) {
+        do {
+            let miscGatewayRequest = MiscGatewayRequest(fingerprintResponse: "{\"threeDS2FingerprintTimeout\":true}")
+            miscGatewayRequest.headers = ["Content-Type": "application/json"]
+            miscGatewayRequest.url = URL
+            let encodedJson = try JSONEncoder().encode(miscGatewayRequest)
+            if let base64Encoded = String(data: encodedJson.base64EncodedData(), encoding: .utf8) {
```

Comment:
Perhaps adding a helper in this `MiscGatewayRequest` object would be a good idea as well, similarly to what we've done on Android